bundle/bundle.go

```diff
@@ -140,35 +140,38 @@ type Action struct {
 
 // ValuesOrDefaults returns parameter values or the default parameter values
 func ValuesOrDefaults(vals map[string]interface{}, b *Bundle) (map[string]interface{}, error) {
-	requiredMap := map[string]struct{}{}
-	for _, key := range b.Parameters.Required {
-		requiredMap[key] = struct{}{}
-	}
 	res := map[string]interface{}{}
 
-	for name, def := range b.Parameters.Fields {
-		s, ok := b.Definitions[def.Definition]
-		if !ok {
-			return res, fmt.Errorf("unable to find definition for %s", name)
+	if b.Parameters != nil {
```

Comment:
It's nit picking, but I prefer the early return notation:
```go
if b.Parameters == nil{
    return res, nil
}
...
```

Reply:
I agree, here it is probably better for code (and PR) readability. Fixed.